File: src/main/java/iteration2/FacultyMember.java
package iteration2;


import com.fasterxml.jackson.annotation.JsonProperty;
import iteration2.Course;
import iteration2.Student;

public abstract class FacultyMember extends Person {

	public FacultyMember(String name, String id) {
		super(name,id);
	}

	public abstract void gradeStudents(Student student, Course course);

	@Override
	public String toString() {
		return "FacultyMember{} " + super.toString();
	}


}

File: FacultyMember.py
from abc import abstractmethod

from Person import Person


class FacultyMember(Person):
    def __init__(self, name: str, id: str):
        super().__init__(name, id)

    @abstractmethod
    def grade_students(self, student, course):
        pass

    def __str__(self) -> str:
        return f"FacultyMember {{}} {super().__str__()}"